dataclasses_avroschema/schema_generator.py

```diff
@@ -115,30 +115,31 @@ def _get_enum_type_map(cls: Type[CT]) -> Dict[str, enum.EnumMeta]:
         return enum_types
 
     @classmethod
-    def _deserialize_list_type(cls: Type[CT], *, field_name: str, payload: Any) -> List:
-        data: List = []
-        for value in payload:
-            if isinstance(value, dict):
-                data.append(cls._deserialize_complex_types(value))
-            elif isinstance(value, list):
-                data.append(cls._deserialize_list_type(field_name=field_name, payload=value))
-            else:
-                data.append(value)
-        return data
+    def _deserialize_list(cls: Type[CT], field: str, payload: Any) -> List:
+        field_def = cls.schema_def.fields_map[field]
+        item_type = field_def.type.__args__[0]  # it could be List[SOMETHING] or Tuple[Something]
```

Comment:
I know that it's dirty.
But i don't know who we could extract the type from type annotation using a different way.